src/utils/proxyMap.ts

```diff
@@ -0,0 +1,111 @@
+import { proxy } from 'valtio'
+
+type KeyValRecord<K, V> = [key: K, value: V]
+
+type InternalProxyMap<K, V> = Map<K, V> & {
+  data: KeyValRecord<K, V>[]
+  toJSON: object
+}
+
+/**
+ * proxyMap
+ *
+ * This is to create a proxy which mimic the native Map behavior.
+ * The API is the same as Map API
+ *
+ * @example
+ * import { proxyMap } from 'valtio/utils'
+ * const state = proxyMap([["key", "value"]])
+ *
+ * //can be used inside a proxy as well
+ * const state = proxy({
+ *   count: 1,
+ *   map: proxyMap()
+ * })
+ *
+ * // When using an object as a key, you can wrap it with `ref` so it's not proxied
+ * // this is useful if you want to preserve the key equality
+ * import { ref } from 'valtio'
+ *
+ * const key = ref({})
+ * state.set(key, "value")
+ * state.get(key) //value
+ *
+ * const key = {}
+ * state.set(key, "value")
+ * state.get(key) //undefined
+ */
+export const proxyMap = <K, V>(
+  entries?: Iterable<readonly [K, V]> | null
+): Map<K, V> => {
+  const map = proxy<InternalProxyMap<K, V>>({
```

Comment:
This typing is different from `proxySet`.
```ts
  const set: InternalProxySet<T> = proxy({
```
Should they be consistent?

Reply:
Indeed. Good remark. I've done the change